pkg/util/log/crash_reporting.go

```diff
@@ -260,29 +290,22 @@ func reportablesToSafeError(depth int, format string, reportables []interface{})
 		file, line, _ = caller.Lookup(depth)
 	}
 
-	if e, ok := reportables[0].(error); ok {
+	if e, ok := reportables[0].(error); ok && len(reportables) == 1 && format == "" {
 		// Special case so that `panic(err)` for a safe `err` returns `err` (and
 		// doesn't wrap it in a `safeError`). In fact, we go further and also try
 		// the errors.Cause() similarly. The effect is that something like
 		// `errors.Wrap(someSafeErr, "gibberish")` can report `someSafeErr` verbatim.
 		for _, err := range []error{e, errors.Cause(e)} {
-			if format == "" && len(reportables) == 1 && redact(err) == err.Error() {
+			if _, safe := redact(err); safe {
 				return err
 			}
 		}
 	}
 
 	redacted := make([]string, 0, len(reportables))
 	for i := range reportables {
-		msg := redact(reportables[i])
-		typ := fmt.Sprintf("<%T>", reportables[i])
-		if msg == "" {
-			redacted = append(redacted, typ)
-		} else if typ == "<log.SafeType>" || typ == "<*log.SafeType>" {
-			redacted = append(redacted, msg)
-		} else {
-			redacted = append(redacted, typ+": "+msg)
-		}
+		s, _ := redact(reportables[i])
```

Comment:
This is a weird design, with two call sites each of which only uses one of the return values. It's getting really hard to follow this code. Do we really need to keep iterating on this in patch releases?

Reply:
If you don't care about getting better sentry error reporting, then no, we don't. But I think we should because a lot of the reports we're getting in sentry are not actionable, and not finding out until next year sets us back.
Re: the design, I'm open for suggestions.

Reply:
I haven't been paying much attention to sentry so I'm not sure how much we need this. As for design, I can't really suggest a better one because I'm having a hard time telling what we're even trying to do here. Can we reduce some conditional logic by e.g. *always* including the type no matter what?

Basically, my confidence that this code is redacting everything that needs to be redacted is being reduced by every change we add to it. We need to prioritize simplicity here, especially for patch releases.